engine.py

```diff
@@ -107,7 +107,27 @@ def get_rook_moves(self, r, c, moves):
 
 
 	def get_queen_moves(self, r, c, moves):
+		"""
+					Calculates all possible queen moves for a given color (light or dark)
+					based on bishop and rook moves.
+
+					input parameter(s):
+					r     --> starting row (int)
+					c     --> starting colum (int)
+					moves --> possible moves container (list)
+
+					return parameter(s):
+					None
+				"""
 		##TODO
+		if self.light_to_move:  # if it's light's turn to move
+			self.get_bishop_moves(r, c, moves)
+			self.get_rook_moves(r, c, moves)
+
+		else: # if it's dark's turn to move
+			self.get_bishop_moves(r, c, moves)
+			self.get_rook_moves(r, c, moves)
+
```

Comment:
You don't need the if condition, since that is already handled in both bishop and rook moves. Just call the two methods